File: 프로그래머스/1/42840. 모의고사/모의고사.py
'''
반복
1번 : 1,2,3,4,5
2번 : 2,1,2,3,2,4,2,5
3번 : 3,3,1,1,2,2,4,4,5,5

'''
def solution(answers):
    answer = []
    p1 = [1,2,3,4,5] * 2000
    p2 = [2,1,2,3,2,4,2,5] * 2000
    p3 = [3,3,1,1,2,2,4,4,5,5] * 2000
    count1 = 0
    count2 = 0
    count3 = 0
    for i in range(len(answers)):
        if answers[i] == p1[i]:
            count1 += 1
        if answers[i] == p2[i]:
            count2 += 1
        if answers[i] == p3[i]:
            count3 += 1
            
    maxcnt = max(count1, count2, count3)
    if maxcnt == count1:
        answer.append(1)
    if maxcnt == count2:
        answer.append(2)
    if maxcnt == count3:
        answer.append(3)
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
class Solution {
    public int[] solution(int[] answers) {
        int[] answer = {};
        int[] test1 = {1,2,3,4,5};
        int[] test2 = {2,1,2,3,2,4,2,5};
        int[] test3 = {3,3,1,1,2,2,4,4,5,5};
        int[] correct = new int[3];
        
        for(int i=0; i<answers.length; i++){
            if(answers[i] == test1[i%test1.length]){
                correct[0]++;
            }
            if(answers[i] == test2[i%test2.length]){
                correct[1]++;
            }
            if(answers[i] == test3[i%test3.length]){
                correct[2]++;
            }
        }
        int max = correct[0];
        int cnt = 1;
        for(int i=1; i<correct.length; i++){
            if(correct[i] == max){
                cnt++;
            }else if(correct[i] > max){
                max = correct[i];
                cnt = 1;
            }
        }
        int[] result = new int[cnt];
        int j=0;
        for(int i=0; i<correct.length; i++){
            if(correct[i] == max){
                result[j++] = i+1;
            }
        }
        
        return result;
    }
}

/*
1번 : 1,2,3,4,5
2번 : 2,1,2,3,2,4,2,5
3번 : 3,3,1,1,2,2,4,4,5,5

개수로 나머지연산

제일많이 맞춘갯수 정해서 그값이랑 같으면 배열에 담기
*/